tools/pipelines/templates/include-upload-npm-logs.yml

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+# Licensed under the MIT License.
+
+# Template to upload logs from npm (not pnpm!) invocations during a job run.
+# These can be useful for debugging npm-related issues
+
+parameters:
+  # Path to npm logs directory.
+  # Defaults to the path that npm ends up using in Azure Pipelines if there are no overrides to its config.
+  # Files that include this template generally shouldn't need to change this.
+  - name: npmLogsDirectory
+    type: string
+    default: '/home/cloudtest/.npm/_logs'
+
+steps:
+- task: Bash@3
+  displayName: Upload npm logs
+  condition: succeededOrFailed()
+  continueOnError: true
+  inputs:
+    targetType: 'inline'
+    workingDirectory: '${{ parameters.npmLogsDirectory }}'
+    script: |
+      set -eu -o pipefail
+      TARFILE_FULLPATH=$(Agent.TempDirectory)/npm-logs.tar
+      # Put all files in a tarball, in case there's more than one
```

Comment:
Is there any chance that this logs directory is empty? Should there be a check for that?

Reply:
Hmm yeah, I guess it might if no npm invocations ocurred during the job and then this runs. I'll add a check.

Reply:
Actually `tar` will just create an empty tarball in that case which I think is fine for this. I'd rather keep the script as simple as possible as long as it doesn't produce wrong results or crashes. Empty folder of npm logs seems like a reasonable result in this case.